### PR TITLE
feat: add select database for Redis related functions

### PR DESCRIPTION
**Pull request by wettper**
Under normal circumstances, Redis deploys and uses more than one database, so this time we added select database related functions, hoping to make some contributions to `apisix`

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by spacewander — COMMENTED** [review]
Need test in https://github.com/apache/apisix/blob/master/t/plugin/limit-count-redis.t

File: t/plugin/limit-count-redis.t [hunk 1]
```diff
@@ -122,7 +122,146 @@ passed
...
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 4: set route, with redis host and port but wrong database
```
Comment (spacewander) [inline t/plugin/limit-count-redis.t:170]: Better to add test case after the last one.
And you need to use another test case to hit the route.

**Review by spacewander — CHANGES_REQUESTED** [review]

File: doc/zh-cn/plugins/limit-count.md [hunk 2]
```diff
@@ -44,6 +44,7 @@
 | redis_host      | string   | `redis` 必须 |         |                                                              | 当使用 `redis` 限速策略时，该属性是 Redis 服务节点的地址。   |
 | redis_port      | integer  | 可选         | 6379    | [1,...]                                                      | 当使用 `redis` 限速策略时，该属性是 Redis 服务节点的端口     |
 | redis_password  | string   | 可选         |         |                                                              | 当使用 `redis` 限速策略时，该属性是 Redis 服务节点的密码。   |
+| redis_database  | integer  | 可选         | 0       | redis_database >= 0                                          | 当使用 `redis` 限速策略时，该属性是 Redis 服务节点中使用的 database，并且只针对 非Redis集群模式（单实例模式 或者 提供单入口的Redis公有云服务）生效。   |
```
Comment (tokers) [inline doc/zh-cn/plugins/limit-count.md:47]: Some minor style problems: Keep empty space on both sides of English words. Remove redundant empty spaces between Chinese words.

Comment (wettper) [reply doc/zh-cn/plugins/limit-count.md]: Uh, yes, I modified the related grammar, thank you very much for reminding

File: t/plugin/limit-count-redis.t (hunk 1 above)
Comment (wettper) [reply t/plugin/limit-count-redis.t]: Uh, ok, I added a unit test file, and I am debugging related errors

**Comment by wettper** [comment]
@spacewander @tokers  Thank you both for your reminders, I have resubmitted...

**Review by spacewander — APPROVED** [review]

**Comment by wettper** [comment]
@spacewander  Sorry, I accidentally clicked the "Re-request review" button ^*^

**Review by tokers — APPROVED** [review]

**Pull request merged**